owncloudApp/src/main/java/com/owncloud/android/ui/activity/DrawerActivity.kt

```diff
@@ -0,0 +1,699 @@
+
+    private var accountChooserToggle: ImageView? = null
+    private var accountEndAccountAvatar: ImageView? = null
+    private var accountMiddleAccountAvatar: ImageView? = null
+
+    private var drawerLayout: DrawerLayout? = null
+    private var drawerLogo: ImageView? = null
+    private var drawerToggle: ActionBarDrawerToggle? = null
+
+    private var navigationView: NavigationView? = null
+
+    private var isAccountChooserActive = false
+    private var checkedMenuItem = Menu.NONE
+
+    /**
+     * accounts for the (max) three displayed accounts in the drawer header.
+     */
+    private var accountsWithAvatars = arrayOfNulls<Account>(3)
+
+    /**
+     * Initializes the drawer and its content.
+     * This method needs to be called after the content view has been set.
+     */
+    protected open fun setupDrawer() {
+
+        // Allow or disallow touches with other visible windows
+        drawer_layout?.filterTouchesWhenObscured = PreferenceUtils.shouldDisallowTouchesWithOtherVisibleWindows(this)
+        // Allow or disallow touches with other visible windows
+        nav_view?.filterTouchesWhenObscured = PreferenceUtils.shouldDisallowTouchesWithOtherVisibleWindows(this)
+        if (nav_view != null) {
+            accountChooserToggle = findNavigationViewChildById(R.id.drawer_account_chooser_toogle) as ImageView?
+            accountMiddleAccountAvatar = findNavigationViewChildById(R.id.drawer_account_middle) as ImageView?
+            accountEndAccountAvatar = findNavigationViewChildById(R.id.drawer_account_end) as ImageView?
+            // Set background header image and logo, if any
+            if (resources.getBoolean(R.bool.use_drawer_background_header)) {
+                (findNavigationViewChildById(R.id.drawer_header_background) as ImageView?)?.setImageResource(R.drawable.drawer_header_background)
+            }
+            if (resources.getBoolean(R.bool.use_drawer_logo)) {
+                drawer_logo?.setImageResource(R.drawable.drawer_logo)
+            }
+
+            accountChooserToggle?.setImageResource(R.drawable.ic_down)
+            isAccountChooserActive = false
+
+            //Notch support
+            nav_view.addOnAttachStateChangeListener(object : View.OnAttachStateChangeListener {
+                override fun onViewAttachedToWindow(v: View) {
+                    if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.P) {
+                        val displayCutout = v.rootWindowInsets.displayCutout
+                        if (displayCutout != null) {
+                            val rlDrawerActiveUser =
+                                findNavigationViewChildById(R.id.drawer_active_user) as ConstraintLayout?
+                            val orientation = resources.configuration.orientation
+                            if (orientation == Configuration.ORIENTATION_PORTRAIT) {
+                                val displayCutoutDP = displayCutout.safeInsetTop /
```

Comment:
Too much logic for a view, I would move it to a viewmodel or, failing that, to a utils file.